### PR TITLE
Ported to Native splash screen

### LINKED ISSUE #138: Use Native Splash Screen
## Discussion :speaking_head:
I'm a participant of GSSOC '21 and wish to open my first issue in this project and the program.

The app currently uses Flutter splash screen with a timer of 3 seconds.
Native splash screen replaces the initial white screen (app initialising time) with our required splash screen.

## Context
This port needs to be done to improve the user's experience as the Flutter splash screen simply wastes the user's time with the 3 second delay which is added time above the already app initialising time.

## Detailed Description
I wish to port the splash screen to native one using the flutter_native_splash package which can generate native splash screen for both android and ios, allowing to use an image and a background that replaces the initial white splash screen without wasting the user's time.

I can work on this and please assign this issue to me.

### PR DESCRIPTION
**Pull request by AJAYK-01**
### Related Issue
- Fixes #138 
### Proposed Changes
- Android and iOS native splash screens added
- Removed entrypoint of exisiting Flutter splash screen in main.dart

### Additional Information
- The splash screens have been added using the [flutter_native_splash](https://pub.dev/packages/flutter_native_splash) package, a dev dependency which is no longer required after creating the splash screens, hence removed.
- No changes in pubspec.yaml and no changes in documentation.

### Checklist
- [x] Tested **(on Android)**
- [x] No Conflicts
- [x] Change In Code
- [ ]  Change In Documentation

### Screenshots
Before this PR            | After this PR
:-----------------------: | :-----------------------:
 |

### CODE REVIEW

**Comment by himanshusharma89** [comment]
@AJAYK-01 Can you tell me about the Runner directory updates? If these are your system generated files you should remove them from this PR. 
And also, can you delete the splash screen as it is not required now.

**Comment by AJAYK-01** [comment]
Runner directory updates are generated by the flutter_native_splash package to update iOS splash screen
It is added but not tested because it requires a Mac to see in action

**Comment by AJAYK-01** [comment]
> @AJAYK-01 Can you tell me about the Runner directory updates? If these are your system generated files you should remove them from this PR. 
> And also, can you delete the splash screen as it is not required now.

Ok i will delete the splash.dart widget

**Comment by himanshusharma89** [comment]
> Runner directory updates are generated by the flutter_native_splash package to update iOS splash screen
> It is added but not tested because it requires a Mac to see in action

I didn't see any packages in the pubspec.yaml in your PR. Also, squash your commits. If you are facing an error, wait for 2-3 min, I am pushing some changes.

**Comment by AJAYK-01** [comment]
> > Runner directory updates are generated by the flutter_native_splash package to update iOS splash screen
> > It is added but not tested because it requires a Mac to see in action
> 
> I didn't see any packages in the pubspec.yaml in your PR. Also, squash your commits. If you are facing an error, wait for 2-3 min, I am pushing some changes.

Yes, there is no change, I removed the package as it is no longer required after creating the splash screen
It is a dev dependency

File: lib/helpers/route_page.dart [hunk 1]
```diff
@@ -8,14 +8,13 @@ import 'package:retro_shopping/views/home.dart';
 import 'package:retro_shopping/views/profile.dart';
 import 'package:retro_shopping/views/search.dart';
 import 'package:retro_shopping/widgets/product/product_page.dart';
-import 'package:retro_shopping/widgets/splash.dart';
 
 class RoutePage {
   static Route<dynamic> generateRoute(RouteSettings settings) {
     switch (settings.name) {
       case '/':
```
Comment (himanshusharma89) [inline lib/helpers/route_page.dart:15]: Remove this case, as we already have a case for Dashboard.

**Review by himanshusharma89 — CHANGES_REQUESTED** [review]

**Comment by himanshusharma89** [comment]
Also, squash the commits: https://github.com/wprig/wprig/wiki/How-to-squash-commits

File: lib/helpers/route_page.dart (hunk 1 above)
Comment (AJAYK-01) [reply lib/helpers/route_page.dart:15]: I've made the case '/' point to dashboard as well.
And doing the squash now

Comment (himanshusharma89) [reply lib/helpers/route_page.dart:15]: @AJAYK-01 there already exists a case for dashboard, so you can remove this case

**Comment by AJAYK-01** [comment]
I have performed the changes and squashed

**Pull request merged**